setup.py

```diff
@@ -1619,7 +1621,11 @@ def detect_multiprocessing(self):
                                    libraries=libs,
                                    include_dirs=["Modules/_multiprocessing"]))
 
+            if HOST_PLATFORM.startswith('linux'):
+                multiprocessing_lib = ['pthread']
```

Comment:
@xdegaye Does it look right to you?

Reply:
Also cc @rossburton.

Reply:
On Android, the Python platform name is ``linux`` and pthread is not a standalone library but it is included in the libc (named ``bionic``), same as on MacOS X.  Therefore ``-lpthread`` is not used at the link stage.

IMO it is confusing to have both configure and setup.py handle the setting of ``-lpthread`` especially since the configure implementation of pthread variants is already quite involved. It would be better to just fix ``configure.ac`` if it is needed.

It is not clear from the comments on [issue 31171](https://bugs.python.org/issue31171) if the compiler ``-pthread`` option is available. If that option is available then running configure with ``ac_cv_pthread=yes`` sets ``CC="$CC -pthread"`` and that should fix the problem raised in the issue. Quoted from gcc documentation:
```
'-pthread'
     Define additional macros required for using the POSIX threads
     library.  You should use this option consistently for both
     compilation and linking.  This option is supported on GNU/Linux
     targets, most other Unix derivatives, and also on x86 Cygwin and
     MinGW targets.
```

Reply:
And if the cross-compiler does not support the ``-pthread`` option then run make with LDFLAGS:
```
    LDFLAGS=-lpthread make
```